File: 136-SingleNumber/136-SingleNumber.js
// Last updated: 3/6/2026, 2:09:34 PM
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    let obj = {};
    for(let char of nums){
        if(obj[char]===undefined){
            obj[char]=1;
        }
        else{
            obj[char]++;
        }
    }

    for(let value in obj){
        if(obj[value]==1){
         return Number(value);
        }
    }
};

File: 136-SingleNumber/136-SingleNumber.cpp
// Last updated: 6/2/2026, 9:25:17 PM
1class Solution {
2public:
3    int singleNumber(vector<int>& nums) {
4        int ans = 0;
5
6        for (int num : nums) {
7            ans ^= num;
8        }
9
10        return ans;
11    }
12};